training/metrics.py

```diff
@@ -27,83 +24,151 @@ def count_trainable_params(model):
   total_trainable_params = int(sum(np.prod(p.shape.as_list()) for p in object_identity.ObjectIdentitySet(weights)))
```

Comment:
You can use `tf.size(p)` (or probably `p.size` ? I don't know, tensorflow documentation is cursed) instead of `np.prod(p.shape.as_list()`
see https://www.tensorflow.org/api_docs/python/tf/size

Reply:
updated accordingly.